django packages: convert to python packaging format + add python3 variant

commodo opened the pull request:
Maintainer: me / @cotequeiroz 
Compile tested: https://github.com/openwrt/openwrt/commit/bc47285cb3c0125424e628521f905f1f0d7b4cef  x86
Run tested: N/A

These changes convert a bunch of django packages to the python packaging format + adding python3 variants.
The aim is to have them ready to be able to switch seafile/seahub to Python3 (at a later point in time).

Signed-off-by: Alexandru Ardelean <ardeleanalex@gmail.com>

commodo commented:
adding
@jefferyto , @BKPepe 
more eyes to take a look in case i goofed something

BKPepe commented on lang/python/django-appconf/Makefile:
```diff
@@ -9,39 +9,58 @@ include $(TOPDIR)/rules.mk
   SECTION:=lang
   CATEGORY:=Languages
   TITLE:=A helper class for handling configuration defaults of packaged apps gracefully.
   URL:=http://django-appconf.readthedocs.org/
```
```suggestion
  URL:=https://django-appconf.readthedocs.io
```

commodo replied:
ack

BKPepe commented on lang/python/django-appconf/Makefile:
```diff
@@ -9,39 +9,58 @@ include $(TOPDIR)/rules.mk
-$(eval $(call BuildPackage,django-appconf))
+$(eval $(call PyPackage,python-django-appconf))
+$(eval $(call BuildPackage,python-django-appconf))
+
```
What about src packages?

commodo replied:
sure;

BKPepe commented on lang/python/django-compressor/Makefile:
```diff
@@ -9,38 +9,68 @@ include $(TOPDIR)/rules.mk
 PKG_SOURCE:=django_compressor-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=https://pypi.python.org/packages/82/76/1355459f90714517c52f264aa7245b52e59a273ec16e8f8d505fa6c342f8/
 PKG_HASH:=9616570e5b08e92fa9eadc7a1b1b49639cce07ef392fc27c74230ab08075b30f
+PKG_MAINTAINER:=Alexandru Ardelean <ardeleanalex@gmail.com>
```
New line, please.

commodo replied:
ack

BKPepe commented on lang/python/django-compressor/Makefile:
```diff
@@ -9,38 +9,68 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=django-compressor
 PKG_VERSION:=2.2
```
The latest one is 2.3.

commodo replied:
ack;
but i would defer updates of packages;
mostly due to the fact that it's some extra unknowns

most packages in the seafile dependency chain require updating;
i will do an update later, when also updating seafile

BKPepe commented on lang/python/django-compressor/Makefile:
```diff
@@ -9,38 +9,68 @@ include $(TOPDIR)/rules.mk
+PKG_RELEASE:=5
 
 PKG_SOURCE:=django_compressor-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=https://pypi.python.org/packages/82/76/1355459f90714517c52f264aa7245b52e59a273ec16e8f8d505fa6c342f8/
```
What about nicer URL? :)

```suggestion
PKG_SOURCE_URL:=https://files.pythonhosted.org/packages/source/d/django-compressor/
```

BKPepe commented on lang/python/django-compressor/Makefile:
```diff
@@ -9,38 +9,68 @@ include $(TOPDIR)/rules.mk
   SECTION:=lang
   CATEGORY:=Languages
   TITLE:=Compress CSS/JS into single cached files
   URL:=http://django-compressor.readthedocs.org/
```
```suggestion
  URL:=https://django-compressor.readthedocs.io
```

BKPepe commented on lang/python/django-formtools/Makefile:
```diff
@@ -17,24 +17,53 @@ PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
   SECTION:=lang
   CATEGORY:=Languages
   TITLE:=High-level abstractions for Django forms
   URL:=https://django-formtools.readthedocs.io/en/latest/
```
```suggestion
  URL:=https://django-formtools.readthedocs.io/
```

BKPepe commented on lang/python/django-jsonfield/Makefile:
```diff
@@ -9,39 +9,58 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=django-jsonfield
 PKG_VERSION:=1.0.1
```
The latest one is 1.2.0

BKPepe commented on lang/python/django-jsonfield/Makefile:
```diff
@@ -9,39 +9,58 @@ include $(TOPDIR)/rules.mk
 PKG_VERSION:=1.0.1
-PKG_RELEASE:=3
+PKG_RELEASE:=4
 PKG_LICENSE:=BSD-3-Clause
```
```suggestion

PKG_LICENSE:=BSD-3-Clause
PKG_LICENSE_FILES:=LICENSE
```

BKPepe commented on lang/python/django-jsonfield/Makefile:
```diff
@@ -9,39 +9,58 @@ include $(TOPDIR)/rules.mk
 PKG_LICENSE:=BSD-3-Clause
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=https://pypi.python.org/packages/e4/b2/a079f0a2218e0eb7892edbf404e0bbfbb281a6bbf06966b775f5142ed159/
```
What about nicer URL? :)

```suggestion
PKG_SOURCE_URL:=https://files.pythonhosted.org/packages/source/d/django-jsonfield/
```

BKPepe commented on lang/python/django-jsonfield/Makefile:
```diff
@@ -9,39 +9,58 @@ include $(TOPDIR)/rules.mk
 
-$(eval $(call BuildPackage,django-jsonfield))
+$(eval $(call PyPackage,python-django-jsonfield))
+$(eval $(call BuildPackage,python-django-jsonfield))
```
No src packages?

BKPepe commented on lang/python/django-picklefield/Makefile:
```diff
@@ -17,31 +17,50 @@ PKG_SOURCE_URL:=https://files.pythonhosted.org/packages/source/d/django-picklefi
 
-$(eval $(call BuildPackage,django-picklefield))
+$(eval $(call PyPackage,python-django-picklefield))
+$(eval $(call BuildPackage,python-django-picklefield))
```
No src packages?

BKPepe commented on lang/python/django-restframework/Makefile:
```diff
@@ -9,7 +9,7 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=django-restframework
 PKG_VERSION:=3.9.0
```
Latest one is 3.9.4.

BKPepe commented on lang/python/django-restframework/Makefile:
```diff
@@ -9,7 +9,7 @@ include $(TOPDIR)/rules.mk
 PKG_VERSION:=3.9.0
-PKG_RELEASE:=3
+PKG_RELEASE:=4
 PKG_LICENSE:=BSD-3-Clause
```
```suggestion 

PKG_LICENSE:=BSD-3-Clause
PKG_LICENSE_FILES:=LICENSE.md
```

BKPepe commented on lang/python/django-picklefield/Makefile:
```diff
@@ -17,31 +17,50 @@ PKG_SOURCE_URL:=https://files.pythonhosted.org/packages/source/d/django-picklefi
 PKG_HASH:=ce7fee5c6558fe5dc8924993d994ccde75bb75b91cd82787cbd4c92b95a69f9c
```
There is already version 2.0. Can you please update it? :)

cotequeiroz commented on lang/python/django-formtools/Makefile:
```diff
@@ -7,7 +7,7 @@ include $(TOPDIR)/rules.mk
-PKG_RELEASE:=4
+PKG_RELEASE:=5
 
 PKG_MAINTAINER:=Eneas U de Queiroz <cote2004-github@yahoo.com>
```
While at it, would you please update my email:
```suggestion
PKG_MAINTAINER:=Eneas U de Queiroz <cotequeiroz@gmail.com>
```
BTW, the old one will remain valid, I'm changing it because I got tired of DMARC headers being added when I post something to the mailing list.

cotequeiroz commented on net/seafile-seahub/Makefile:
```diff
@@ -24,16 +24,15 @@ include ../../lang/python/python-package.mk
-	+django-picklefield +django-postoffice +django-restframework \
+  DEPENDS:=+python +pillow \
 	+django-simple-captcha +django-statici18n +django-webpack-loader \
 	+flup +gunicorn +openpyxl \
```
You'll need to increase `PKG_RELEASE`, since this changes the install-time dependencies, due to the package renames.  Also, you'll need to adjust to #9328 (I'm counting on it being merged soon, since it is a trivial fix).

commodo commented:
Changelog v1 -> v2:
* renamed `define Package/python-django-xxxx/Default`  -> `define Package/django-xxxx/Default`
* updated email `cotequeiroz@gmail.com` in `django-formtools`
* use `$(PKG_NAME)` in build rules as much as possible; reasoning: it makes things super easy to diff and identify mistakes with all repetitive changes; this helped with:
   * added PKG_LICENSE_FILES where missing
   * all currently updated packages (in this PR) have (roughly) the same ordering/format of PKG fields
   * updated URLs to python-hosted site
   * update URLs to readthedocs.io (where the case)

commodo commented:
updated with PR comments;
I did not update any packages; I can do that if is insisted upon; but I would generally defer this to later;

commodo commented:
ping

BKPepe commented on lang/python/django-jsonfield/Makefile:
```diff
@@ -9,39 +9,62 @@ include $(TOPDIR)/rules.mk
   TITLE:=JSONField for django models
-  URL:=https://github.com/bradjasper/django-jsonfield
-  DEPENDS:=+python +python-django
+  URL:=https://github.com/dmkoch/django-jsonfield
```
I think the right one is: https://github.com/adamchainz/django-jsonfield

commodo replied:
oh; right
thanks

BKPepe commented on net/seafile-seahub/Makefile:
```diff
@@ -9,7 +9,7 @@ include $(TOPDIR)/rules.mk
-PKG_RELEASE:=5
+PKG_RELEASE:=6
 PKG_LICENSE:=Apache-2.0
 
```
Add please PKG_LICENSE_FILES, PKG_MAINTAINER.

commodo replied:
I thought I was maintainer here;
hmm; weird; well, it's Monday and the coffee is just kicking in

BKPepe commented on lang/python/django-postoffice/Makefile:
```diff
@@ -9,40 +9,64 @@ include $(TOPDIR)/rules.mk
   SUBMENU:=Python
   SECTION:=lang
   CATEGORY:=Languages
   TITLE:=A Django app to monitor and send mail asynchronously, complete with template support.
```
This one is too long so it isn't shown properly in `make menuconfig`. While you are it, the dot is not necessary.

BKPepe commented on lang/python/django-appconf/Makefile:
```diff
@@ -9,39 +9,62 @@ include $(TOPDIR)/rules.mk
   SUBMENU:=Python
   SECTION:=lang
   CATEGORY:=Languages
   TITLE:=A helper class for handling configuration defaults of packaged apps gracefully.
```
This one is too long so it isn't shown properly in `make menuconfig`. While you are it, the dot is not necessary.

commodo commented:
[hmm, maybe marking comments as resolved was too soon; i'll un-mark them]

Changelog v2 -> v3:
* updated URL for django-jsonfiel
* seafile-seahub: added myself as maintainer + PKG_LICENSE_FILES
* reduced titles for django-{appconf,postoffice}

cotequeiroz commented:
I've run-tested them with seafile; all in order, so I'm merging this.
